### PR TITLE
fix: add mfa reference docs

### PR DESCRIPTION
**Pull request by kangmingtay**
## What kind of change does this PR introduce?
* Add MFA methods to reference docs

### CODE REVIEW

File: spec/common-client-libs-sections.json [hunk 1]
```diff
@@ -577,6 +577,46 @@
...
+        "id": "mfa-verify",
+        "title": "Verify a challenge",
+        "slug": "auth-mfa-verify",
+        "product": "auth",
+        "type": "function",
+        "libs": ["js"]
+      },
+      {
+        "id": "mfa-challenge-and-verify",
+        "title": "Verify a challenge",
```
Comment (dannykng) [inline spec/common-client-libs-sections.json:607]: Should this be:
```suggestion
        "title": "Create and verify a challenge",
```

File: spec/supabase_js_v2_temp_new_shape.yml [hunk 2]
```diff
@@ -453,6 +453,87 @@ functions:
             if (event == 'PASSWORD_RECOVERY') console.log('PASSWORD_RECOVERY', session)
           })
           ```
+  - id: mfa-enroll
+    title: 'mfa.enroll()'
+    $ref: '@supabase/gotrue-js.GoTrueMFAApi.enroll'
+    notes: |
+      - Currently, `totp` is the only supported `factorType`.
+      - To generate a QR code containing the `totp` secret,
```
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:461]: Incomplete sentence?

File: spec/supabase_js_v2_temp_new_shape.yml [hunk 3]
```diff
@@ -453,6 +453,87 @@ functions:
...
+      - Currently, `totp` is the only supported `factorType`.
+      - To generate a QR code containing the `totp` secret, 
+      - To verify the factor next, see `auth.mfa.challengeAndVerify()`.
+      - To generate a QR code for the `totp` secret in nextjs, you can do the following:
+      ```html
+      <Image src={data.totp.qr_code} alt={data.totp.uri} layout="fill"></Image>
+      ```
+    examples:
+      - id: enroll-totp-factor
+        name: Enroll a Time-based One time password (TOTP) factor
```
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:469]:
```suggestion
        name: Enroll a time-based, one-time password (TOTP) factor
```

File: spec/supabase_js_v2_temp_new_shape.yml [hunk 4]
```diff
@@ -453,6 +453,87 @@ functions:
...
+          const { data, error } = await supabase.auth.mfa.enroll({ 
+            factorType: 'totp' 
+          })
+          ```
+  - id: mfa-challenge
+    title: 'mfa.challenge()'
+    $ref: '@supabase/gotrue-js.GoTrueMFAApi.challenge'
+    examples:
+      - id: create-mfa-challenge
+        name: Creates a challenge for a factor
```
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:482]:
```suggestion
        name: Create a challenge for a factor
```

File: spec/supabase_js_v2_temp_new_shape.yml [hunk 5]
```diff
@@ -453,6 +453,87 @@ functions:
...
+          const { data, error } = await supabase.auth.mfa.challenge({ 
+            factorId: '34e770dd-9ff9-416c-87fa-43b31d7ef225'
+          })
+          ```
+  - id: mfa-verify
+    title: 'mfa.verify()'
+    $ref: '@supabase/gotrue-js.GoTrueMFAApi.verify'
+    examples:
+      - id: verify-challenge
+        name: Verifies a challenge for a factor
```
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:495]:
```suggestion
        name: Verify a challenge for a factor
```

File: spec/supabase_js_v2_temp_new_shape.yml [hunk 6]
```diff
@@ -453,6 +453,87 @@ functions:
...
+          })
+          ```
+  - id: mfa-challenge-and-verify
+    title: 'mfa.challengeAndVerify()'
+    $ref: '@supabase/gotrue-js.GoTrueMFAApi.challengeAndVerify'
+    notes: |
+      - Combines `mfa.challenge()` and `mfa.verify()` in 1 step
+    examples:
+      - id: challenge-and-verify
+        name: Creates and verifies a challenge for a factor
```
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:512]:
```suggestion
        name: Create and verify a challenge for a factor
```

File: spec/supabase_js_v2_temp_new_shape.yml [hunk 7]
```diff
@@ -453,6 +453,87 @@ functions:
...
+            challengeId: '4034ae6f-a8ce-4fb5-8ee5-69a5863a7c15', 
+            code: '123456'
+          })
+          ```
+  - id: mfa-get-authenticator-assurance-level
+    title: 'mfa.getAuthenticatorAssuranceLevel()'
+    $ref: '@supabase/gotrue-js.GoTrueMFAApi.getAuthenticatorAssuranceLevel'
+    notes: |
+      - Authenticator Assurance Level (AAL) is the measure of the strength of an authentication mechanism.
+      - In Supabase, having an AAL of `aal1` typically refers to having the 1st factor of authentication such as an email and password or oauth sign-in while `aal2` refers to the 2nd factor of authentication such as Time-based one-time-password (TOTP).
```
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:527]:
```suggestion
      - In Supabase, having an AAL of `aal1` refers to having the 1st factor of authentication such as an email and password or OAuth sign-in while `aal2` refers to the 2nd factor of authentication such as a time-based, one-time-password (TOTP).
```

File: spec/supabase_js_v2_temp_new_shape.yml [hunk 8]
```diff
@@ -732,7 +813,31 @@ functions:
             { phone_confirm: true }
           )
           ```
-
+  - id: mfa-list-factors
+    title: 'mfa.listFactors()'
+    $ref: '@supabase/gotrue-js.GoTrueAdminMFAApi.listFactors'
+    examples:
+      - id: list-factors
+        name: List all factors associated to a user
```
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:821]: For consistency:
```suggestion
        name: List all factors for a user
```

File: spec/supabase_js_v2_temp_new_shape.yml [hunk 9]
```diff
@@ -732,7 +813,31 @@ functions:
...
+        js: |
+          ```js
+          const { data, error } = await supabase.auth.admin.mfa.listFactors()
+          ```
+  - id: mfa-delete-factor
+    title: 'mfa.deleteFactor()'
+    $ref: '@supabase/gotrue-js.GoTrueAdminMFAApi.deleteFactor'
+    examples:
+      - id: delete-factor
+        name: Deletes a factor associated to a user
```
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:832]:
```suggestion
        name: Delete a factor for a user
```

**Review by dannykng — APPROVED** [review]

**Comment by MildTomato** [comment]
Screenshot 2022-12-13 at 12 31 50 PM

make sure Docs has built preview before merging

**Comment by MildTomato** [comment]
Screenshot 2022-12-13 at 12 46 56 PM

we also need to be vigilent about these enrichment changes.
make sure they are actually new updated ones when they change

**Review by MildTomato — APPROVED** [review]

**Comment by MildTomato** [comment]
Whats going on with these enrichments? 
@kangmingtay are you running `make` in the `/spec` directory ? or copying things over manually ?

**Review by MildTomato — CHANGES_REQUESTED** [review]
blocked merging while clarifying enrichment issue

**Review by MildTomato — APPROVED** [review]

**Comment by MildTomato** [comment]
snuck missing env vars for ticket site in

**Pull request merged**